test/TensorFlowNET.UnitTest/Training/GradientDescentOptimizerTests.cs

```diff
@@ -67,6 +68,51 @@ public void TestBasic()
             TestBasic<double>();
         }
 
+        private void TestMinimizeResourceVariable<T>() where T : struct
+        {
+            var dtype = GetTypeForNumericType<T>();
+
+            // train.GradientDescentOptimizer is V1 only API.
+            tf.Graph().as_default();
+            using (var sess = self.cached_session())
+            {
+                var var0 = tf.Variable(new[,] { { 1.0f, 2.0f } }, dtype: dtype);
+                var var1 = tf.Variable(new[] { 3.0 }, dtype: dtype);
+                var x = tf.constant(new[,] { { 4.0f }, { 5.0f } }, dtype: dtype);
+
+                var pred = math_ops.matmul(var0, x) + var1;
+                var loss = pred * pred;
+                var sgd_op = tf.train.GradientDescentOptimizer(3.0f).minimize(loss);
+
+                var global_variables = tf.global_variables_initializer();
+                sess.run(global_variables);
+
+                sess.run(new[] { var0, var1 });
+                // Fetch params to validate initial values
+                self.assertAllCloseAccordingToType<T>(new[,] { { 1.0, 2.0 } }, self.evaluate<T[,]>(var0));
+                self.assertAllCloseAccordingToType(new[] { 3.0 }, self.evaluate<T[]>(var1));
+                // Run 1 step of sgd
+                sgd_op.run();
+                // Validate updated params
+                var np_pred = 1.0 * 4.0 + 2.0 * 5.0 + 3.0;
+                var np_grad = 2 * np_pred;
+                self.assertAllCloseAccordingToType(
```

Comment:
@Wanglongzhi2001 Hm, now it calculates but the test doesn't pass. However, the code corresponds to TensorFlow original test. I have to check math there.

Reply:
There just was a small typo, but I didn't have time to debug it :-D